default.nix

```diff
@@ -1,6 +1,13 @@
 {
 # Compiler in a form ghc8101 == GHC 8.10.1, just remove spaces and dots
-  compiler    ? "ghc883"
+#  2020-07-05: By default using default GHC for Nixpkgs, see https://search.nixos.org/packages?query=ghc&from=0&size=500&channel=unstable for current version (currently ghc883 == GHC 8.8.3)
+  compiler    ? "ghc884"
+
+# Deafult.nix is a unit package abstraciton that allows to abstract over packages even in monorepos:
+# Example: pass --arg cabalName --arg packageRoot "./subprojectDir", or map default.nix over a list of tiples for subprojects.
+# cabalName is package resulting name: by default and on error resolves in haskellPackages.developPackage to project root directory name by default, but outside the haskellPackages.developPackage as you see below packageRoot can be different
+, cabalName ? "hnix"
```

Comment:
I don't understand the motivation for having this parameter. Could you give an example where it would be useful?

Reply:
It is a ground work towards the support of monorepo with several projects.

Since `develoPackage` should include `name` of the Cabal file, and different projects have different Cabal names, passing through the `cabalName` and `projectRoot` - allows to traverse over the projects in monorepo.

Since HNix plans to move to monorepo at some near point - I preserved this ground work, there is also CI builds which matrix traverse over project roots `[ ./ ]`, aka singleton currently.

Reply:
> Since HNix plans to move to monorepo at some near point

Who is planning this? Is there an issue where this plan is discussed? If not, we should have one.

Reply:
It is `"Split parser & types out into different packages"` #328, there the mororepo approach was discussed much, several people voted up for it, John also up for completely split the lib from executable, what is to do - is of course to what is most effective.

Reply:
Thanks for the link. I think it would be good to have a separate thread for the mono-repo discussion. It's currently rather hidden in a very long thread.